react/ImageManager.js

```diff
@@ -0,0 +1,46 @@
+import React, { Component } from 'react'
+import PropTypes from 'prop-types'
+
+import Image from './Image'
+import MagnifyingImage from './MagnifyingImage'
+import ThumbnailList from './ThumbnailList'
+
+const FIRST_ELEMENT = 0
+
+class ImageManager extends Component {
+  constructor(props) {
+    super(props)
+    this.state = {
+      selectedImage: this.props.images[FIRST_ELEMENT]
+    }
+  }
+
+  handleThumbnailClick(image) {
+    this.setState({
+      selectedImage: image
+    })
+  }
+  
+  render() {
+    const { images, zoom, thumbnail, thumbnailPosition } = this.props
+    const { selectedImage } = this.state
+
+    return (
+      <div className="ma1 w-50 dt dt--fixed">
+        { thumbnail ? <ThumbnailList position={thumbnailPosition} images={images} onThumbnailClick={this.handleThumbnailClick.bind(this)} /> : '' }
```

Comment:
Try to use the arrow notation to bind functions. Something like `handleThumbnailClick => (image) {}`, instead of binding inside the render function.